### ISSUE TITLE
Enums can't be iterated over in Python 3.10

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1352]
**Describe the bug**
`Enum`s do not support iteration in Python 3.10.

**To Reproduce**
Set the `pythonVersion` to `3.10` in pyrightconfig.json and type check the snippet below.

**Expected behavior**
That no error would be produced.

**Screenshots or Code**

```python
from enum import Enum, auto

class Foo(Enum):
    bar = auto()

# "Type[Foo]" is not iterable
#   "__iter__" method not defined
for member in Foo:
    ...

```

**VS Code extension or command-line**
Either, 1.1.99

**Comment by layday** [comment]
They also seem to be missing `__name__`:

```py
# Cannot access member "__name__" for type "Type[Foo]"
#   Member "__name__" is unknown
Foo.__name__
```

... so perhaps this is not specific to iteration.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a strange bug that had little or nothing to do with enums. It has to do with the strange circular dependency between `type` and `object`. The `type` class derives from `object`, but the `object` class uses `type` as a metaclass. We've gotten lucky in the past that these have always been resolved in an order that worked correctly, but a small change in the typeshed stubs for Python 3.10 caused this resolution order to break under certain circumstances. The result was that `object` had an unresolvable metaclass, which effectively disabled all metaclass processing for all classes. Since `__iter__` is provided by the `EnumMeta` metaclass, this logic was being skipped.

I've added special logic to handle the circular dependency in a deterministic manner. This will be in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
